pkg/provider/aws/secretsmanager/secretsmanager_test.go

```diff
@@ -534,6 +543,53 @@ func TestSetSecret(t *testing.T) {
 				err: nil,
 			},
 		},
+		"SetSecretSucceedsWithExistingSecretAndKMSKeyAndDescription": {
+			reason: "a secret can be pushed to aws secrets manager when it already exists",
+			args: args{
+				store: makeValidSecretStore().Spec.Provider.AWS,
+				client: fakesm.Client{
+					GetSecretValueWithContextFn: fakesm.NewGetSecretValueWithContextFn(secretValueOutput, nil),
```

Comment:
```suggestion
					GetSecretValueWithContextFn: fakesm.NewGetSecretValueWithContextFn(secretValueOutput, &getSecretCorrectErr),
```

This will make it use the actual metadata. But unfortunately, there is no way to access the metadata or see what it was parsed as. Basically, you need to update the fake thing to be able to check it.

Reply:
done